File: src/jlppc/regimys/evolution/E_TradeItem.java
package jlppc.regimys.evolution;

import jlppc.regimys.objects.Pokemon;
import jlppc.regimys.objects.items.Item;

public class E_TradeItem extends Evolution {
	
	protected Item forEvolve;
	
	
	public E_TradeItem(int evo, Item forEvolve) {
		super(evo);
		this.forEvolve = forEvolve;
	}

	@Override
	public boolean checkEvolve(Pokemon toCheck) {
		if(forEvolve == null){
			return false;
		}
		return toCheck.itemHeld().equals(forEvolve);
	}
}

File: src/jlppc/regimys/evolution/ETradeItem.cpp

#include "evolutions.hpp"


namespace Evolutions {
	bool E_TradeItem::checkEvolve(OpMon const &toCheck) {
		if (forEvolve == NULL) {
			return false;
		}
		Item *held = toCheck.itemHeld();
		return held == (forEvolve);
	}
}

